website/static/js/quickProjectSearchPlugin.js

```diff
@@ -0,0 +1,551 @@
+/**
+ * UI and function to quick search projects
+ */
+
+var $ = require('jquery');
+var m = require('mithril');
+var $osf = require('js/osfHelpers');
+
+// CSS
+require('css/quick-project-search-plugin.css');
+
+// XHR config for apiserver connection
+var xhrconfig = function(xhr) {
+    xhr.withCredentials = true;
+};
+
+
+var quickSearchProject = {
+    controller: function() {
+        var self = this;
+        self.nodes = m.prop([]); // Pending nodes waiting to be displayed
+        self.displayedNodes = m.prop([]); // Nodes that are rendered
+        self.nonMatchingNodes = m.prop([]); //Nodes that don't match search query
+        self.sortState = m.prop('dateDesc');
```

Comment:
The comments after these helps a lot. Can you add comments at the end of all of them?